pkg/controller/storagecluster/initialization_reconciler_test.go

```diff
@@ -452,6 +452,24 @@ func assertExpectedExternalResources(t assert.TestingT, reconciler ReconcileStor
 			}
 		}
 	}
+	// get the ceph cluster
+	request.Name = generateNameForCephCluster(sc)
+	foundCeph := &cephv1.CephCluster{}
+	err = reconciler.client.Get(nil, request.NamespacedName, foundCeph)
+	assert.NoError(t, err)
+	// set the state to 'ClusterStateConnected' (to mock a successful external cluster connection)
+	foundCeph.Status.State = cephv1.ClusterStateConnected
+	err = reconciler.client.Update(nil, foundCeph)
+	assert.NoError(t, err)
+	// call 'ensureNoobaaSystem()' to make sure it takes appropriate action
+	// when ceph cluster is connected to an external cluster
+	err = reconciler.ensureNoobaaSystem(sc, reconciler.reqLogger)
+	assert.NoError(t, err)
+	fNoobaa := &nbv1.NooBaa{}
+	request.Name = "noobaa"
+	// expectation is to get an appropriate Noobaa object
+	err = reconciler.client.Get(nil, request.NamespacedName, fNoobaa)
+	assert.NoError(t, err)
```

Comment:
This does not belong here. It should be tested in `noobaa_system_reconciler_test.go`, either as a new case in `TestEnsureNooBaaSystem()` or as it's own test function. And we need to test both a connected external CephCluster and an external CephCluster that is not connected yet.

Reply:
Yes @jarrpa , I have created a separate test under `noobaa_system_reconciler_test.go` file, named `TestNoobaaSystemInExternalClusterMode`. Please take a look...

@umangachapagain , @obnoxxx ^^^